pkg/rclgo/wait.go

```diff
@@ -200,7 +200,11 @@ func (w *WaitSet) Run(ctx context.Context) (err error) {
 			return err
 		}
 		if rc := C.rcl_wait(&w.rcl_wait_set_t, -1); rc != C.RCL_RET_OK {
-			return errorsCast(rc)
+			if rc == C.RCL_RET_TIMEOUT {
```

Comment:
maybe best to have something like:

```
int timeout = -1
if rc := C.rcl_wait(&w.rcl_wait_set_t, timeout); rc != C.RCL_RET_OK {
			if (timeout < 0 && rc == C.RCL_RET_TIMEOUT) {
```

this way if the timeout ever gets changed to be _not_ infinite, the timeout will keep functioning.

Reply:
@jbcpollak The timeout occurring with a finite timeout value is even more ok to continue on, per https://docs.ros2.org/beta1/api/rcl/wait_8h.html#a732278988c802fe2c9d8ec24752f9dd9 . Timing out with a value of -1 is weird, but it certainly appears to fix our issue.

Reply:
oh, I misunderstood the code / didn't read far enough - I thought this was re-waiting, not continuing. I get it now, sounds fine.

Reply:
this construct:

```
if rc := C.rcl_wait(&w.rcl_wait_set_t, -1); rc != C.RCL_RET_OK {
			if rc == C.RCL_RET_TIMEOUT {
```

would probably make more sense as a `switch` now though, but whatever

Reply:
how so? If it's OK we want to execute the rest of the loop body. If it's a timeout, we want to continue from the top of the loop. If it's a non-timeout error, we want to return the error and exit the loop. I'm not sure how a switch statement would make it more readable.

Reply:
@jbcpollak i changed it to a switch/case. Please :eye: again